src/server/game/Entities/Player/Player.cpp

```diff
@@ -27594,4 +27594,124 @@ bool Player::IsPetDismissed()
     return false;
 }
 
+/* Premium character interaction.
+ * These functions will allow modules to access the new generic
+ * premium tables designed to help modules that interact with
+ * players and characters without the need to create other tables.
+ */
+int Player::GetCharacterPremiumLevel(uint32 guid)
+{
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_PREMIUM_CHARACTER_PREMIUM_LEVEL);
+    stmt->setUInt32(0, GUID_LOPART(guid));
+
+    PreparedQueryResult result = CharacterDatabase.Query(stmt);
+
+    if (!result)
+        return 0;
+
+    int premium_level = (*result)[0].GetInt8();
+    return premium_level;
+}
+
+
+bool Player::CreateCharacterPremiumLevel(uint32 guid, int premiumLevel)
+{
+    // Validate if character to be inserted already has a premium level
+    int hasPremiumLevel = GetCharacterPremiumLevel(guid);
+
+    if (!hasPremiumLevel)
```

Comment:
If you can re-set the premium level from 4 to 2 you should name the command maybe "premium account set ACC LEVEL"
Because you can do it more than just once, I suppose?

Reply:
Well, you actually can't because if you try to create and there's a premium level already, you first need to delete. I made that intentional i don't know if it makes sence